packages/core/core/src/requests.js

```diff
@@ -0,0 +1,377 @@
+// @flow strict-local
+
+import type {FilePath} from '@parcel/types';
+import type WorkerFarm from '@parcel/workers';
+import type {
+  AssetRequestDesc,
+  AssetRequestResult,
+  Dependency,
+  ParcelOptions,
+  TransformationOpts
+} from './types';
+import type RequestTracker, {RequestRunnerAPI} from './RequestTracker';
+import type AssetGraph from './AssetGraph';
+import type ParcelConfig from './ParcelConfig';
+import type {TargetResolveResult} from './TargetResolver';
+import type {EntryResult} from './EntryResolver'; // ? Is this right
+
+import invariant from 'assert';
+import nullthrows from 'nullthrows';
+import path from 'path';
+import {isGlob} from '@parcel/utils';
+import {nodeFromAssetGroup} from './AssetGraph';
+import ResolverRunner from './ResolverRunner';
+import {EntryResolver} from './EntryResolver';
+import TargetResolver from './TargetResolver';
+import {RequestRunner, generateRequestId} from './RequestTracker';
+
+export type AssetGraphBuildRequest =
+  | EntryRequest
+  | TargetRequest
+  | AssetRequest
+  | DepPathRequest;
+
+type EntryRequest = {|
+  id: string,
+  +type: 'entry_request',
+  request: FilePath,
+  result?: EntryResult
+|};
+
+type TargetRequest = {|
+  id: string,
+  +type: 'target_request',
+  request: FilePath,
+  result?: TargetResolveResult
+|};
+
+type AssetRequest = {|
+  id: string,
+  +type: 'asset_request',
+  request: AssetRequestDesc,
+  result?: AssetRequestResult
+|};
+
+type DepPathRequest = {|
+  id: string,
+  +type: 'dep_path_request',
+  request: Dependency,
+  result?: AssetRequestDesc
+|};
+
+export class EntryRequestRunner extends RequestRunner<FilePath, EntryResult> {
```

Comment:
We could probably extract these into their own files at some point...

Reply:
Yeah, I'm imagining the runners will get their own files, it was just easier to put all of this logic in one file while I was trying out different things. I can move them into different files once we start taking on more use cases than building the AssetGraph.